reproject/mosaicking.py

```diff
@@ -0,0 +1,199 @@
+        # smallest resolution specified.
+        scales = proj_plane_pixel_scales(wcs)
+        resolutions.append(np.min(np.abs(scales)))
+
+    # We now stack the coordinates - however the ICRS class can't do this
+    # so we have to use the high-level SkyCoord class.
+    corners = SkyCoord(corners)
+    references = SkyCoord(references)
+
+    # If no reference coordinate has been passed in for the final header, we
+    # determine the reference coordinate as the mean of all the reference
+    # positions. This choice is as good as any and if the user really cares,
+    # they can set  it manually.
+    if reference is None:
+        reference = SkyCoord(references.data.mean(), frame=references.frame)
+
+    # In any case, we need to convert the reference coordinate (either specified
+    # or automatically determined) to the requested final frame.
+    reference = reference.transform_to(frame)
+
+    # Determine resolution if not specified
+    if resolution is None:
+        resolution = np.min(resolutions) * u.deg
+
+    # Determine the resolution in degrees
+    cdelt = resolution.to(u.deg).value
+
+    # Construct WCS object centered on position
+    wcs_final = celestial_frame_to_wcs(frame, projection=projection)
+
+    rep = reference.represent_as('unitspherical')
+    wcs_final.wcs.crval = rep.lon.degree, rep.lat.degree
+    wcs_final.wcs.cdelt = -cdelt, cdelt
+
+    # For now, set crpix to (1, 1) and we'll then figure out where all the images
+    # fall in this projection, then we'll adjust crpix.
+    wcs_final.wcs.crpix = (1, 1)
+
+    # Find pixel coordinates of all corners in the final WCS projection. We use origin=1
+    # since we are trying to determine crpix values.
+    xp, yp = skycoord_to_pixel(corners, wcs_final, origin=1)
+
+    if auto_rotate:
+
+        # Use shapely to represent the points and find the minimum rotated rectangle
+        from shapely.geometry import MultiPoint
+        mp = MultiPoint(list(zip(xp, yp)))
+
+        # The following returns a list of rectangle vertices - in fact there are
+        # 5 coordinates because shapely represents it as a closed polygon with
+        # the same first/last vertex.
+        xr, yr = mp.minimum_rotated_rectangle.exterior.coords.xy
+        xr, yr = xr[:4], yr[:4]
+
+        # The order of the vertices is not guaranteed to be constant so we
+        # take the vertices with the two smallest y values (which, for a
+        # rectangle, guarantees that the vertices are neighboring)
+        order = np.argsort(yr)
+        x1, y1, x2, y2 = xr[order[0]], yr[order[0]], xr[order[1]], yr[order[1]]
+
+        # Determine angle between two of the vertices. It doesn't matter which
+        # ones they are, we just want to know how far from being straight the
+        # rectangle is.
+        angle = np.arctan2(y2 - y1, x2 - x1)
+
+        # Determine the smallest angle that would cause the rectangle to be
+        # lined up with the axes.
+        angle = angle % 90
```

Comment:
`np.arctan2` returns something in units of radians, not degrees, right?

Reply:
unless x1, x2, y1, and y2 are quantities?

Reply:
Good catch!

Reply:
They aren't quantities - this was a real bug